cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -0,0 +1,125 @@
+        'environment': {'cloud_network': prov_data['cloud_network'],
+                        'cloud_tenant': prov_data['cloud_tenant']},
+        'properties': {'instance_type': partial_match(flavor_name)},
+    }
+
+    instance_name = prov_form_data['catalog']['vm_name']
+
+    try:
+        instance = provider.appliance.collections.cloud_instances.create(
+            instance_name,
+            provider,
+            prov_form_data, find_in_cfme=True
+        )
+
+    except KeyError:
+        # some yaml value wasn't found
+        pytest.skip('Unable to find an image map in provider "{}" provisioning data: {}'
+                    .format(provider, prov_data))
+
+    yield instance
+
+    instance.cleanup_on_provider()
+
+
+@pytest.mark.ignore_stream('5.9')
+def test_create_instance_with_zero_disk_flavor(new_instance, soft_assert):
+    view = navigate_to(new_instance, 'Details')
+    prov_data = new_instance.provider.data['provisioning']
+    power_state = view.entities.summary('Power Management').get_text_of('Power State')
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = view.entities.summary('Relationships').get_text_of('VM Template')
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    flavors = [f.name for f in new_instance.provider.mgmt.api.flavors.list()]
+    flavor = view.entities.summary('Relationships').get_text_of('Flavor')
+    soft_assert(flavor in flavors)
+
+    # Assert other relationships in a loop
+    props = [('Availability Zone', 'availability_zone'),
+             ('Cloud Tenants', 'cloud_tenant'),
+             ('Virtual Private Cloud', 'cloud_network')]
+
+    for p in props:
+        v = view.entities.summary('Relationships').get_text_of(p[0])
+        soft_assert(v == prov_data[p[1]])
+
+
+def test_flavor_crud(appliance, provider, request):
+    collection = appliance.collections.cloud_flavors
+    flavor = collection.create(name=fauxfactory.gen_alpha(),
+                               provider=provider,
+                               ram=RAM,
+                               vcpus=VCPUS,
+                               disk=DISK_SIZE,
+                               swap=SWAP_SIZE,
+                               rxtx=RXTX)
+
+    view = appliance.browser.create_view(navigator.get_class(collection, 'All').VIEW)
+    view.flash.assert_success_message(
+        'Add of Flavor "{}" was successfully initialized.'.format(flavor.name))
+
+    wait_for(lambda: flavor.exists, delay=5, timeout=600, fail_func=flavor.refresh,
+             message='Wait for flavor to appear')
+
+    flavor.delete()
+    view = appliance.browser.create_view(navigator.get_class(collection, 'All').VIEW)
+    view.flash.assert_success_message(
+        'Delete of Flavor "{}" was successfully initiated.'.format(flavor.name))
+
+    wait_for(lambda: not flavor.exists, delay=5, timeout=600, fail_func=flavor.refresh,
+             message='Wait for flavor to appear')
+
+    @request.addfinalizer
```

Comment:
yep - just the kind of thing I was looking to see for cleanup!

Now, its placement in the test function matters too - Please move this up to right after the call to `collection.create`, before the `view` is even set in order to check flash messages.

Defining the method there won't execute it right away - `request` will just register the method and save it for the teardown phase of test execution.  Then it'll get run after the test itself finishes (pretty much where you've defined it here).  With the method defined at the bottom, if say the call to `appliance.browser.create_view` results in an exception, the cleanup method won't be registered yet, and won't run - leaving a new 'test' flavor behind on the provider.